temporal/api/cloud/cloudservice/v1/service.proto

```diff
@@ -105,6 +105,29 @@ service CloudService {
         };
     }
 
+    // Failover a multi-region namespace
+    rpc FailoverNamespace (FailoverNamespaceRequest) returns (FailoverNamespaceResponse) {
+        option (google.api.http) = {
+            post: "/api/v1/namespaces/{namespace}/failover",
+            body: "*"
+        };
+    }
+
+    // Get failover history on a namespace
+    rpc GetNamespaceFailoverHistory (GetNamespaceFailoverHistoryRequest) returns (GetNamespaceFailoverHistoryResponse) {
+        option (google.api.http) = {
+            get: "/api/v1/namespaces/{namespace}/failover-history",
+        };
+    }
+
+    // Add a new region to a namespace
+    rpc AddNamespaceRegion (AddNamespaceRegionRequest) returns (AddNamespaceRegionResponse) {
```

Comment:
Another option: Use UpdateNamespace to support this case.

Reply:
@atihkin Is this ok to you?

Reply:
can we keep this capabaility in AddNamespaceRegion instead?

Reply:
Synced with @atihkin. We agreed to have this new API for two reasons:
1. Better user experience.
2. The add region is special and a longer async operation which is different than update namespace.

Reply:
For automation use cases, the primary consumers of these apis, have a patch only endpoint adds complexity on the client. It will need to selectively call a AddNamespaceRegion() for updating the regions, why calling a UpdatingNamespace to update all else. 

We can have a additional patch endpoint. But i strongly believe UpdateNamespace should also support adding a region.

Reply:
Also do we plan to add support for removing a region? anytime in the future.

Reply:
Yes. in long term we will support.

Reply:
My concern is the UpdateNamespace with new region and other fields together would be hard to deal with. Any good option we can limit to only update one field at a time?